src/os/ui/menu/menu.js

```diff
@@ -131,7 +131,7 @@ os.ui.menu.Menu.prototype.onClick_ = function(e) {
       openerClicked = !!jqEl[0];
     }
 
-    if (openerClicked && os.ui.menu.Menu.supportsToggle(openerEl)) {
+    if (openerClicked && (os.ui.menu.Menu.supportsToggle(openerEl) || this.supportsToggle(openerEl))) {
```

Comment:
I'm not certain we want the menu itself to determine toggle behavior, because the same menu could potentially be opened from different DOM elements.

It might be preferable to allow a DOM element to identify itself as a menu toggle element. For example, we could add a `js-menu__toggle` class to the element and look for that in `os.ui.menu.Menu.supportsToggle`. That would both avoid any breaking changes (nothing would reasonably have a class like this already) while introducing new functionality.

Reply:
I did think about the multiple use of the menu, but it seemed like the common case was single use, and the next most common case was the opener was button (which is already handled). If you have the `openerEl` then you could figure it out for the corner cases.

I hadn't consider adding a class to the opener element. Will give that a try.